envi/archs/arm/const.py

```diff
@@ -16,49 +15,66 @@
 REV_ARMv5J  =   0b0000000000000000100000 #       32        0x20
 REV_ARMv5TE =   0b0000000000000001000000 #       64        0x40
 REV_ARMv6   =   0b0000000000000010000000 #      128        0x80
-REV_ARMv6T2 =   0b0000000000000100000000 #      256        0x100
-REV_ARMv6M  =   0b0000000000001000000000 #      512        0x200
+REV_ARMv6M  =   0b0000000000000100000000 #      256        0x100
+REV_ARMv6T2 =   0b0000000000001000000000 #      512        0x200
 REV_ARMv7A  =   0b0000000000010000000000 #     1024        0x400
 REV_ARMv7R  =   0b0000000000100000000000 #     2048        0x800
 REV_ARMv7M  =   0b0000000001000000000000 #     4096        0x1000
 REV_ARMv7EM =   0b0000000010000000000000 #     8192        0x2000
 REV_ARMv8A  =   0b0000000100000000000000 #    16384        0x4000
 REV_ARMv8R  =   0b0000001000000000000000 #    32768        0x8000
 REV_ARMv8M  =   0b0000010000000000000000 #    65536        0x10000
-REVS_ARMv4  = (REV_ARMv4 | REV_ARMv4T)
-REVS_ARMv5  = (REV_ARMv5 | REV_ARMv5T | REV_ARMv5E | REV_ARMv5J | REV_ARMv5TE)
-REVS_ARMv6  = (REV_ARMv6 | REV_ARMv6T2 | REV_ARMv6M)
-REVS_ARMv7  = (REV_ARMv7A | REV_ARMv7R | REV_ARMv7M | REV_ARMv7EM) 
-REVS_ARMv8  = (REV_ARMv8A | REV_ARMv8R | REV_ARMv8M)
-REV_ALL = (REVS_ARMv4 | REVS_ARMv5 | REVS_ARMv6 | REVS_ARMv7 | REVS_ARMv8)
-
-#Not sure, did from memory , needs to be confirmed
-REVT_THUMB16 = (REVS_ARMv5 | REVS_ARMv6)
-REVT_THUMB2  = (REVS_ARMv7 | REVS_ARMv8)
-REVT_THUMBEE = (REVS_ARMv7 | REVS_ARMv8)
-
-#In progress - Draft version
-ARCH_REVS = {
-    0:  ['ARMv4',     REV_ARMv4],
-    1:  ['ARMv4T',    REV_ARMv4T],
-    2:  ['ARMv5',     REV_ARMv5],
-    3:  ['ARMv5T',    REV_ARMv5T],
-    4:  ['ARMv5E',    REV_ARMv5E],
-    5:  ['ARMv5J',    REV_ARMv5J],
-    6:  ['ARMv5TE',   REV_ARMv5TE],
-    7:  ['REV_ARMv6', REV_ARMv6],
-    8:  ['ARMv6T2',   REV_ARMv6T2],
-    9:  ['ARMv6M',    REV_ARMv6M],
-    10: ['ARMv7A',    REV_ARMv7A],
-    11: ['ARMv7R',    REV_ARMv7R],
-    12: ['ARMv7M',    REV_ARMv7M],
-    13: ['ARMv7EM',   REV_ARMv7EM],
-    14: ['ARMv8A',    REV_ARMv8A],
-    15: ['ARMv8R',    REV_ARMv8R],
-    16: ['ARMv8M',    REV_ARMv8M],
-    17: ['ARMALL',    REV_ALL]
-}
-ARCH_REVSLEN = len(ARCH_REVS) 
+
+#To be replaced with dynamic setting up to prevent bugs when adding additional versions above
+REV_ALL_ARMv4  = (REV_ARMv4 | REV_ARMv4T)
+REV_ALL_ARMv5  = (REV_ARMv5 | REV_ARMv5T | REV_ARMv5E | REV_ARMv5J | REV_ARMv5TE)
+REV_ALL_ARMv6  = (REV_ARMv6 | REV_ARMv6T2 | REV_ARMv6M)
+REV_ALL_ARMv7  = (REV_ARMv7A | REV_ARMv7R | REV_ARMv7M | REV_ARMv7EM) 
+REV_ALL_ARMv8  = (REV_ARMv8A | REV_ARMv8R | REV_ARMv8M)
+
+#Set below
+REV_ARM_ALL = 0
```

Comment:
Done and updated tests that use it too.

Date: Wed, 8 Jun 2016 11:47:19 -0700
From: notifications@github.com
To: vivisect@noreply.github.com
CC: redheadedrod@hotmail.com; author@noreply.github.com
Subject: Re: [atlas0fd00m/vivisect] Unittest (#5)

In envi/archs/arm/const.py:

> -    14: ['ARMv8A',    REV_ARMv8A],
> -    15: ['ARMv8R',    REV_ARMv8R],
> -    16: ['ARMv8M',    REV_ARMv8M],
> -    17: ['ARMALL',    REV_ALL]
> -}
> -ARCH_REVSLEN = len(ARCH_REVS) 
> +
> +#To be replaced with dynamic setting up to prevent bugs when adding additional versions above
> +REV_ALL_ARMv4  = (REV_ARMv4 | REV_ARMv4T)
> +REV_ALL_ARMv5  = (REV_ARMv5 | REV_ARMv5T | REV_ARMv5E | REV_ARMv5J | REV_ARMv5TE)
> +REV_ALL_ARMv6  = (REV_ARMv6 | REV_ARMv6T2 | REV_ARMv6M)
> +REV_ALL_ARMv7  = (REV_ARMv7A | REV_ARMv7R | REV_ARMv7M | REV_ARMv7EM) 
> +REV_ALL_ARMv8  = (REV_ARMv8A | REV_ARMv8R | REV_ARMv8M)
> +
> +#Set below
> +REV_ARM_ALL = 0

can this be "REV_ALL" or "REV_ALL_ARM" ?  since it's a combination, it makes sense to keep the "REV_ALL*" naming.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.